src/ui/page/share/dataset/steps/serviceStorageSelection/serviceStorageSelection.tsx

```diff
@@ -32,20 +35,38 @@ export const ServiceStorageSelection: FC = () => {
   const [selectedService, setSelectedService] = useState<DataverseItemDetails | null>(null)
   const currentLng = activeLanguageWithDefault().lng
   const selectedServiceRef = useRef<HTMLDivElement | null>(null)
+  const dispatchNotification = useDispatchNotification()
 
-  const { loadDataverse, dataverse, setByTypeFilter, isLoading, hasNext, setLanguage } =
-    useDataverseStore(state => ({
-      loadDataverse: state.loadDataverse,
-      dataverse: state.dataverse,
-      isLoading: state.isLoading,
-      hasNext: state.hasNext,
-      setLanguage: state.setLanguage,
-      setByTypeFilter: state.setByTypeFilter
-    }))
+  const {
+    loadDataverse,
+    dataverse,
+    setByTypeFilter,
+    byPropertyFilter,
+    setByPropertyFilter,
+    isLoading,
+    hasNext,
+    setLanguage,
+    error
+  } = useDataverseStore(state => ({
+    loadDataverse: state.loadDataverse,
+    dataverse: state.dataverse,
+    isLoading: state.isLoading,
+    hasNext: state.hasNext,
+    setLanguage: state.setLanguage,
+    byPropertyFilter: state.byPropertyFilter,
+    setByPropertyFilter: state.setByPropertyFilter,
+    setByTypeFilter: state.setByTypeFilter,
+    error: state.error
+  }))
 
-  const handleServiceSearch = useCallback((searchTearm: string) => {
-    console.log(searchTearm)
-  }, [])
+  const handleServiceSearch = useCallback(
+    (searchTearm: string) => {
+      setSelectedService(null)
+      setByPropertyFilter({ property: 'title', value: searchTearm })()
+      loadDataverse()()
+    },
+    [loadDataverse, setByPropertyFilter]
+  )
```

Comment:
Even though we are aborting results of previous calls, it might be better to debounce the service search function in order to reduce the API calls frequency.

Reply:
I'd actually thought about it and tested it, the problem being that, since the component is controlled, the value to be displayed in the search bar is returned by the parent. 
Its value is `byPropertyFilter()()`, which is calculated by `handleServiceSearch` (which calls `setByPropertyFilter`).
As a result, when debouncing, the letters typed by the user take a long time to appear in the search bar.

Reply:
Yes, for a controlled component, you don't have the choice of passing the parent's value. This value can come from a local state of the parent, which doesn't need to be synchronised with `byPropertyFilter()()`. I think it's completely fine for our case.

We can address this later on.